components/organisms/ViewportProvider.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
```

Comment:
You don't need `React` to be loaded in all modules anymore.
Also, prefer importing only what you use.

```git
-import React from 'react';
+import { createContext, useEffect, useState } from 'react';
```